src/lib/y2storage/filesystems/btrfs.rb

```diff
@@ -123,8 +185,108 @@ def self.default_btrfs_subvolume_path
         Yast::ProductFeatures.GetStringFeature(section, feature)
       end
 
+      # Updates the list of subvolumes for the Btrfs filesystem mounted at root
+      #
+      # Subvolumes are shadowed or unshadowed according to current mount points
+      # in the whole system.
+      #
+      # @see #shadow_btrfs_subvolumes
+      # @see #unshadow_btrfs_subvolumes
+      #
+      # @param devicegraph [Devicegraph]
+      def self.refresh_root_subvolumes_shadowing(devicegraph)
+        filesystem = BlkFilesystem.all(devicegraph).detect { |f| f.root? && f.is?(:btrfs) }
+        return if filesystem.nil?
+
+        filesystem.shadow_btrfs_subvolumes(devicegraph)
+        filesystem.unshadow_btrfs_subvolumes(devicegraph)
+      end
+
+      # Removes current shadowed subvolumes
+      # Only subvolumes that "can be shadowed" will be removed.
```

Comment:
Why don't use a more explicit names like `remove_shadowed_subvols` and `restore_unshadowed_subvols`?